### ISSUE TITLE
Determine main entities in services

### ISSUE THREAD

**Issue by zingfeng** [plug-service/crm#1]
In a CRM microservice, there are several entities that should be managed, including:

Customers: This entity represents the individuals or businesses who use the CRM system to manage their interactions with the organization.

Contacts: This entity represents the individuals or stakeholders within a customer's organization who are responsible for the interactions with the organization.

Leads: This entity represents potential customers who have shown interest in the organization's products or services.

Opportunities: This entity represents potential sales deals that are being pursued by the organization.

Accounts: This entity represents the business accounts that are associated with the organization's customers.

Activities: This entity represents the interactions between the organization and its customers, such as phone calls, emails, meetings, and tasks.

Products: This entity represents the products or services that are offered by the organization.

When building a CRM microservice, it's important to identify which entities are relevant to your organization's specific needs and to ensure that the microservice can manage and store data related to these entities efficiently and securely.

**Comment by zingfeng** [comment]
? Should contain entity `campaign marketing`?

**Comment by vanruc** [comment]
> ? Should contain entity `campaign marketing`?

Yes, we should. Even need to implement it very detail. Customer will need to monitor their budget for campaign marketing.
They also need to analysis what is the most profitable marketing channel. And how their staff using budget...

How about customer 360 and products/services 360?

**Comment by zingfeng** [comment]
1. Can you list entities related to `campaign marketing` and `budget`?
2. The purpose here is to list all entities and divide them into several microservices. I mean microservice should not contain too many entities.  Regarding CRM, I guess we will have about 3 microservices
3. 
> How about customer 360 and products/services 360?

You mean we should define entities based on the enitites of  `customer 360`? Please give me the link of its documentation, it will help me understand more
https://www.salesforce.com/products/

**Comment by zingfeng** [comment]
## CRM
I think we can split CRM into 2 main microservices:
1. **crm-customer**: Manage 6 entities containing customer or potential customer information
- Lead
- Customers
- Accounts
- Contact
- Activities
- Opportunities

2. **crm-campaign**: Manage entities containing **marketing campaigns** and its related entities such as history using budget, results of campaign ... 

## Moreover

1. CRM (customer relationship management) should contain the relationship between the `customer` and `products`
This information about history of the `customer's order`  will be saved in entity `activities` of **crm-customer** and entity `order` in microservice **sale**. Data is duplicated here, and we accept it
